src/tiled/terrainbrush.cpp

```diff
@@ -385,16 +378,25 @@ void TerrainBrush::updateBrush(QPoint cursorPos, const QVector<QPoint> *list)
         int leftID = i - 1;
         int rightID = i + 1;
 
-        if((Map::Orientation)mMapOrientation==Map::Staggered){
-            if(y%2==0){
-                bottomID -= 1;
-                leftID -= layerWidth;
-                rightID = rightID + (layerWidth - 1);
-            }
-            else{
-                upperID += 1;
-                leftID = leftID - (layerWidth -1);
-                rightID += layerWidth;
+        QPoint upPoint(p.x(),p.y()-1);
+        QPoint bottomPoint(p.x(),p.y()+1);
+        QPoint leftPoint(p.x()-1,p.y()-1);
+        QPoint rightPoint(p.x()+1,p.y());
+
+        if(mapDocument() && mapDocument()->map() && mapDocument()->map()->orientation() == Map::Staggered){
```

Comment:
You don't need the `mapDocument() && mapDocument()->map()` part, because `updateBrush` should never be called when there is no current map document, and a `MapDocument` instance _always_ has a map.

Just a minor thing about the code style, please put spaces in `if (` and `) {` (this applies to other parts of the patch as well).